src/main/java/fr/sncf/osrd/speedcontroller/generators/MarecoAllowanceGenerator.java

```diff
@@ -232,17 +234,23 @@ protected double getFirstGuess() {
                 continue;
             }
             // beginning of accelerating slope
-            if (naturalAcceleration > 0 && previousAcceleration < 0) {
-                currentAcceleratingSlope.acceleration = naturalAcceleration;
-                currentAcceleratingSlope.previousAcceleration = previousAcceleration;
-                currentAcceleratingSlope.beginPosition = location.getPathPosition();
+            if (naturalAcceleration > 0) {
+                currentAcceleratingSlope.acceleration += naturalAcceleration;
+                counter += 1;
+                if (previousAcceleration < 0) {
+                    currentAcceleratingSlope.previousAcceleration = previousAcceleration;
+                    currentAcceleratingSlope.beginPosition = location.getPathPosition();
+                }
             } else if (naturalAcceleration < 0
                     && previousAcceleration > 0
                     && currentAcceleratingSlope.acceleration != 0) {
                 // end of accelerating slope
                 currentAcceleratingSlope.endPosition = previousPosition;
                 currentAcceleratingSlope.targetSpeed = speed;
+                currentAcceleratingSlope.acceleration /= counter;
```

Comment:
well that's weird, how does this work?

Reply:
I'm calculating a mean value of the acceleration during all the phase when it's positive, so at each point where the acceleration is positive i add it, and at the end i divide it by the counter. Now that i think about it it would be smart to weight the sum with the distance of each step maybe...